script/bump-version.js

```diff
@@ -0,0 +1,180 @@
+      case 'stable':
+        version = semver.inc(version, 'patch')
+        break
+      default:
+        throw new Error('Invalid bump type.')
+    }
+  } else {
+    throw new Error(`Invalid current version: ${version}`)
+  }
+  return version
+}
+
+// update VERSION file with latest release info
+async function updateVersion (version) {
+  const versionPath = path.resolve(__dirname, '..', 'VERSION')
+  await writeFile(versionPath, version, 'utf8')
+}
+
+// update package metadata files with new version
+async function updatePackageJSON (version) {
+  ['package.json', 'package-lock.json'].forEach(async fileName => {
+    const filePath = path.resolve(__dirname, '..', fileName)
+    const file = require(filePath)
+    file.version = version
+    await writeFile(filePath, JSON.stringify(file, null, 2))
+  })
+}
+
+// update CFBundle version information and overwrite pre-existing file
+// TODO(codebytere): provide these version fields at GN build time
+async function updateInfoPlist (version) {
+  const filePath = path.resolve(__dirname, '..', 'atom', 'browser', 'resources', 'mac', 'Info.plist')
+  const xml = plist.parse(await readFile(filePath, { encoding: 'utf8' }))
+  const file = JSON.parse(JSON.stringify(xml))
+
+  file.CFBundleVersion = version
+  file.CFBundleShortVersionString = version
+
+  await writeFile(filePath, plist.build(file))
+}
+
+// push bump commit to release branch
+async function tagVersion (version) {
+  const gitDir = path.resolve(__dirname, '..')
+  const gitArgs = ['commit', '-a', '-m', `Bump v${version}`, '-n']
+  await GitProcess.exec(gitArgs, gitDir)
+}
+
+// updates atom_version.h file with new semver values
+// TODO(codebytere): auto-generate this
+async function updateVersionH (components, pre) {
+  const filePath = path.resolve(__dirname, '..', 'atom', 'common', 'atom_version.h')
+  const data = await readFile(filePath, 'utf8')
+  const arr = data.split('\n')
+  arr.forEach((item, idx) => {
+    if (item.includes('#define ATOM_MAJOR_VERSION')) {
+      item = `#define ATOM_MAJOR_VERSION ${components[0]}`
+      arr[idx + 1] = `#define ATOM_MINOR_VERSION ${components[1]}`
+      arr[idx + 2] = `#define ATOM_PATCH_VERSION ${components[2]}`
+      arr[idx + 4] = pre ? `#define ATOM_PRE_RELEASE_VERSION ${pre}` : '// #define ATOM_PRE_RELEASE_VERSION'
+    }
+  })
+  await writeFile(filePath, arr.join('\n'))
+}
+
+// updates atom.rc file with new semver values
+async function updateWinRC (components) {
+  const filePath = path.resolve(__dirname, '..', 'atom', 'browser', 'resources', 'win', 'atom.rc')
+  const data = await readFile(filePath, 'utf8')
+  const arr = data.split('\n')
+  arr.forEach((line, idx) => {
+    if (line.includes('FILEVERSION')) {
+      arr[idx] = ` FILEVERSION ${components.join(',')}`
+      arr[idx + 1] = ` PRODUCTVERSION ${components.join(',')}`
+    } else if (line.includes('FILEVERSION')) {
+      arr[idx] = `            VALUE "FileVersion", "${components.slice(0, 3).join('.')}"`
```

Comment:
Should this `else if` be `"FileVersion"`? Otherwise the arms contain the same conditional.